lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -137,21 +137,16 @@ def decrypt_ejson(key_dir)
     end
 
     def fetch_private_key_from_secret
-      secret = run_kubectl_json("get", "secret", EJSON_KEYS_SECRET)
+      out, err, st = @kubectl.run("get", "secret", EJSON_KEYS_SECRET, output: "json")
+      raise EjsonSecretError, err unless st.success?
+      result = JSON.parse(out)
+      secret = result.fetch('items', result)
       encoded_private_key = secret["data"][public_key]
       unless encoded_private_key
         raise EjsonSecretError, "Private key for #{public_key} not found in #{EJSON_KEYS_SECRET} secret"
       end
 
       Base64.decode64(encoded_private_key)
     end
-
-    def run_kubectl_json(*args)
-      args += ["--output=json"]
```

Comment:
There's not really a reason for this to be its own method here since the exception handling can be dealt with in the caller and `Kubectl#run` handles the rest.